cmd/xl-v1-healing-common.go

```diff
@@ -252,39 +252,39 @@ func xlHealStat(xl xlObjects, partsMetadata []xlMetaV1, errs []error) HealObject
 // calculating blake2b checksum.
 func disksWithAllParts(onlineDisks []StorageAPI, partsMetadata []xlMetaV1, errs []error, bucket, object string) ([]StorageAPI, []error, error) {
 	availableDisks := make([]StorageAPI, len(onlineDisks))
-	for index, onlineDisk := range onlineDisks {
+	for diskIndex, onlineDisk := range onlineDisks {
 		if onlineDisk == nil {
 			continue
 		}
 		// disk has a valid xl.json but may not have all the
 		// parts. This is considered an outdated disk, since
 		// it needs healing too.
-		for _, part := range partsMetadata[index].Parts {
+		for partIndex, part := range partsMetadata[diskIndex].Parts {
 			// compute blake2b sum of part.
 			partPath := filepath.Join(object, part.Name)
-			hash := newHash(partsMetadata[index].Erasure.Algorithm)
+			hash := newHash(partsMetadata[diskIndex].Erasure.Checksum[partIndex].Algorithm)
```

Comment:
It looks like replacing `newHash(partsMetadata[diskIndex].Erasure.Checksum[partIndex].Algorithm)` with
`newHash(partsMetadata[diskIndex].Erasure.GetCheckSumInfo(part.Name).Algorithm)` is enough to fix the problem. Can you check if my suggestion would work ? then we'll take another look on the PR.